src/main/java/uk/gov/hmcts/darts/arm/service/impl/CleanupArmResponseFilesServiceImpl.java

```diff
@@ -0,0 +1,230 @@
+import uk.gov.hmcts.darts.common.entity.ObjectRecordStatusEntity;
+import uk.gov.hmcts.darts.common.entity.UserAccountEntity;
+import uk.gov.hmcts.darts.common.enums.ExternalLocationTypeEnum;
+import uk.gov.hmcts.darts.common.helper.CurrentTimeHelper;
+import uk.gov.hmcts.darts.common.repository.ExternalLocationTypeRepository;
+import uk.gov.hmcts.darts.common.repository.ExternalObjectDirectoryRepository;
+import uk.gov.hmcts.darts.common.repository.ObjectRecordStatusRepository;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Objects.nonNull;
+import static uk.gov.hmcts.darts.arm.util.ArchiveConstants.ArchiveRecordOperationValues.ARM_FILENAME_SEPARATOR;
+import static uk.gov.hmcts.darts.arm.util.ArchiveConstants.ArchiveResponseFileAttributes.ARM_CREATE_RECORD_FILENAME_KEY;
+import static uk.gov.hmcts.darts.arm.util.ArchiveConstants.ArchiveResponseFileAttributes.ARM_INPUT_UPLOAD_FILENAME_KEY;
+import static uk.gov.hmcts.darts.arm.util.ArchiveConstants.ArchiveResponseFileAttributes.ARM_INVALID_LINE_FILENAME_KEY;
+import static uk.gov.hmcts.darts.arm.util.ArchiveConstants.ArchiveResponseFileAttributes.ARM_UPLOAD_FILE_FILENAME_KEY;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.ARM_RESPONSE_CHECKSUM_VERIFICATION_FAILED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.ARM_RESPONSE_MANIFEST_FAILED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.ARM_RESPONSE_PROCESSING_FAILED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.STORED;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class CleanupArmResponseFilesServiceImpl implements CleanupArmResponseFilesService {
+
+    private final ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+
+    private final ObjectRecordStatusRepository objectRecordStatusRepository;
+    private final ExternalLocationTypeRepository externalLocationTypeRepository;
+
+    private final ArmDataManagementApi armDataManagementApi;
+    private final UserIdentity userIdentity;
+    private final ArmDataManagementConfiguration armDataManagementConfiguration;
+    private final CurrentTimeHelper currentTimeHelper;
+
+    private final List<String> armResponseFilesToBeDeleted = new ArrayList<>();
+
+    private ObjectRecordStatusEntity storedStatus;
+    private ObjectRecordStatusEntity failedArmResponseManifestFileStatus;
+    private ObjectRecordStatusEntity failedArmResponseProcessing;
+    private ObjectRecordStatusEntity failedArmResponseChecksum;
+    private UserAccountEntity userAccount;
+
+
+    @Override
+    public void cleanupResponseFiles() {
+        initialisePreloadedObjects();
+
+        ExternalLocationTypeEntity armLocation = externalLocationTypeRepository.getReferenceById(ExternalLocationTypeEnum.ARM.getId());
+        List<ObjectRecordStatusEntity> statuses = List.of(storedStatus,
```

Comment:
I think it would be better if we didn't need to populate these class fields - we don't really need to do the `findById` below as we just want the status id's for the query. So I think using the `getReferenceById` in the same way as armLocation is fine as we're not doing anything with the entity e.g. logging description or whatever was being done in the other service.

It can be simplified further by using `.id in (statusIds)` in the `findByStatusInAndStorageLocationAndResponseCleanedAndLastModifiedDateTimeBefore` query.